### PR TITLE
compile is now deprecated and removed completely Gradle 7+

### PR DESCRIPTION
**Pull request by vladspatariu**
### Description

Our CW Roxana Halati caught this while playing around with feature flags in Java. 

### Requirement checklist

- [ ] I have validated my changes on a test/local environment.
- [ ] I have checked the SNYK/Dependabot reports and applied the suggested changes.
- [ ] (Optional) I have updated outdated packages.

### CODE REVIEW

File: website/docs/sdk-reference/java.md [hunk 1]
```diff
@@ -20,10 +20,14 @@ Maven:
   <version>[7.0.0,)</version>
 </dependency>
 ```
-Gradle:
+Old Gradle versions:
```
Comment (z4kn4fein) [inline website/docs/sdk-reference/java.md:23]: I think the new version is enough, we can remove the `compile` one

File: website/docs/sdk-reference/java.md [hunk 2]
```diff
@@ -20,10 +20,14 @@ Maven:
...
 ```bash
 compile group: 'com.configcat', name: 'configcat-java-client', version: '7.+'
 ```
+Gradle 7 and above:
+```bash
+implementation group: 'com.configcat', name: 'configcat-java-client', version: '7.+'
```
Comment (z4kn4fein) [inline website/docs/sdk-reference/java.md:29]:
```suggestion
implementation "com.configcat:configcat-java-client:7.+"
```

Maybe the shorthand version is a bit better, I'm going to change this in the java repo's examples & docs as well

**Review by z4kn4fein — CHANGES_REQUESTED** [review]

**Comment by z4kn4fein** [comment]
Thanks for this heads-up!

**Comment by sigewuzhere** [comment]
:pray: Nice catch!

**Review by z4kn4fein — APPROVED** [review]

**Pull request merged**